src/managers/organization/authentication-by-peers.ts

```diff
@@ -12,16 +12,72 @@ import {
 } from '../../config/errors';
 import { sendMail } from '../../connectors/sendinblue';
 import { updateUserOrganizationLink } from '../../repositories/organization/setters';
-import { getOrganizationsByUserId } from './main';
+import { getOrganizationById, getOrganizationsByUserId } from './main';
 import {
   createModeration,
   findPendingModeration,
 } from '../../repositories/moderation';
 import {
+  DEFAULT_MEMBER_COUNT_THRESHOLD_TO_ACTIVATE_SPONSORSHIP,
   NOTIFY_ALL_MEMBER_LIMIT,
   SUPPORT_EMAIL_ADDRESS,
 } from '../../config/env';
 
+export const isEligibleToSponsorship = async ({
+  id,
+}: Organization): Promise<boolean> => {
+  const organization = await getOrganizationById(id);
+  if (!organization) {
+    return false;
+  }
+
+  const internalActiveUsers = await getInternalActiveUsers(id);
```

Comment:
I suggest to throw another 404 here, if there no internalActiveUsers:
```suggestion
  const internalActiveUsers = await getInternalActiveUsers(id);
  if (internalActiveUsers.lengh < 1) throw Error(`No internal user found for id "${id}".`);
```
Or if you don't want to throw an error, to return false here, because, `return internalActiveUsers.length > memberCountThreshold;` would always be false when internalActiveUsers is 0.

Reply:
yes nice ! Here is what I added.

```
  if (isEmpty(internalActiveUsers)) {
    return false;
  }
  ```